File: 7-reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {

    let negative = x < 0;

    x = Math.abs(x).toString();

    let reversed = [];

    for(let i = x.length - 1; i >= 0; i--){
        reversed.push(x[i]);
    }

    let ans = Number(reversed.join(""));

    if(ans > 2147483647){
        return 0;
    }

    return negative ? -ans : ans;
};

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:

        sign = 1
        if x < 0:
            sign = -1
            x = -x
        x = str(x)
        result = ""
        for i in range(len(x)-1, -1, -1):
            result += x[i]
        ans = int(result) * sign
        if ans < -2**31 or ans > 2**31 - 1:
            return 0
        return ans